Client/Frontend/Library/LibraryViewController.swift

```diff
@@ -97,11 +97,12 @@ class LibraryViewController: UIViewController {
                 }
 
                 if index < panelDescriptors.count {
-                    let panel = self.panelDescriptors[index].viewController
-                    let navigationController = self.panelDescriptors[index].navigationController
-                    let accessibilityLabel = self.panelDescriptors[index].accessibilityLabel
-                    setupLibraryPanel(panel, accessibilityLabel: accessibilityLabel)
-                    self.showPanel(navigationController)
+                    panelDescriptors[index].setup()
```

Comment:
You need to check if this needs to be called. Or... better yet, make `setup()` bail out if the `viewController` is already inited.

Reply:
oops! fixing...